File: algorithms/Codility-master/Codility-master/5_PrefixSums/GenomicRangeQuery.js

// 100%

function solution(S, P, Q) {
    // write your code in JavaScript (Node.js 8.9.4)
    var impact = {'A': 1,
                'C': 2,
                'G': 3,
                'T': 4
    };
    var N = [];
    var f;
    for (let idx =0; idx < P.length; idx++){
        for ( f in impact){
            if ((S.slice(P[idx],Q[idx]+1).indexOf(f) !== -1)){
                N.push(impact[f]);
                break
            }
        }
            
    }
    return N;
}

File: algorithms/Codility-master/Codility-master/5_PrefixSums/GenomicRangeQuery.py

# 100%
def solution(S, P, Q):
    # write your code in Python 3.6
    
    impact = { 'A' : 1,
               'C' : 2,               
               'G' : 3,               
               'T' : 4  }

    N = []
    for idx in range (0 , len(P)):
        for f in impact:
            if f in S[P[idx]:Q[idx]+1]:
                N.append(impact[f])
                print(idx)
                print(f)
                print(S[P[idx]:Q[idx]+1])
                break
    
    return N

print(solution('CAGCCTA', [2, 5, 0], [4, 5, 6]))